packages/functional-tests/tests/subscription-tests/subscription.spec.ts

```diff
@@ -68,10 +68,25 @@ test.describe('severity-2 #smoke', () => {
         'amplitude.subPaySetup.submit',
         'amplitude.subPaySetup.success',
       ];
-      const actualEventTypes = metricsObserver.rawEvents.map((event) => {
-        return event.type;
-      });
-      expect(actualEventTypes).toMatchObject(expectedEventTypes);
+
+      // Added as part of FXA-9322 to resolve flaky bug
+      // Get record of only "with-account" event types as customer should already be logged in
+      const actualEventTypes = metricsObserver.rawEvents
+        .map((event) => {
+          if (event.checkoutType === 'with-account') {
+            return event.type;
+          }
+        })
+        .filter(Boolean);
```

Comment:
On L227 a call to metricsObserver.rawEvents is also made, should that also be filtered for 'with-account'?

Reply:
It looks like that test is for new customer and signs existing customer out on L184-L186 before proceeding. In this scenario, it would be checking for 'without-account'.